Source/Diagnostics/ReducedDiags/DifferentialLuminosity.cpp

```diff
@@ -0,0 +1,286 @@
+            amrex::ParticleReal * const AMREX_RESTRICT u2z = soa_2.m_rdata[PIdx::uz];
+
+            // Extract low-level data
+            auto const n_cells = static_cast<int>(bins_1.numBins());
+
+            // Loop over cells
+            amrex::ParallelFor( n_cells,
+                [=] AMREX_GPU_DEVICE (int i_cell) noexcept
+            {
+                // The particles from species1 that are in the cell `i_cell` are
+                // given by the `indices_1[cell_start_1:cell_stop_1]`
+                index_type const cell_start_1 = cell_offsets_1[i_cell];
+                index_type const cell_stop_1  = cell_offsets_1[i_cell+1];
+                // Same for species 2
+                index_type const cell_start_2 = cell_offsets_2[i_cell];
+                index_type const cell_stop_2  = cell_offsets_2[i_cell+1];
+
+                for(index_type i_1=cell_start_1; i_1<cell_stop_1; ++i_1){
+                    for(index_type i_2=cell_start_2; i_2<cell_stop_2; ++i_2){
+
+                        index_type j_1 = indices_1[i_1];
+                        index_type j_2 = indices_2[i_2];
+
+                        Real u1_square =  u1x[j_1]*u1x[j_1] + u1y[j_1]*u1y[j_1] + u1z[j_1]*u1z[j_1];
+                        Real gamma1 = std::sqrt(1. + u1_square/c2);
+                        Real u2_square = u2x[j_2]*u2x[j_2] + u2y[j_2]*u2y[j_2] + u2z[j_2]*u2z[j_2];
+                        Real gamma2 = std::sqrt(1. + u2_square/c2);
+                        Real u1_dot_u2 = u1x[j_1]*u2x[j_2] + u1y[j_1]*u2y[j_2] + u1z[j_1]*u2z[j_2];
+
+                        // center of mass energy
+                        Real E_com = c2 * std::sqrt(m1*m1 + m2*m2 + 2*m1*m2* (gamma1*gamma2 - u1_dot_u2/c2));
+
+                        // determine particle bin
+                        int const bin = int(Math::floor((E_com-bin_min)/bin_size));
+
+                        if ( bin<0 || bin>=num_bins ) { return; } // discard if out-of-range
+
+                        Real v1_minus_v2_x = u1x[j_1]/gamma1 - u2x[j_2]/gamma2;
+                        Real v1_minus_v2_y = u1y[j_1]/gamma1 - u2y[j_2]/gamma2;
+                        Real v1_minus_v2_z = u1z[j_1]/gamma1 - u2z[j_2]/gamma2;
+                        Real v1_minus_v2_square = v1_minus_v2_x*v1_minus_v2_x + v1_minus_v2_y*v1_minus_v2_y + v1_minus_v2_z*v1_minus_v2_z;
+
+                        Real u1_cross_u2_x = u1y[j_1]*u2z[j_2] - u1z[j_1]*u2y[j_2];
+                        Real u1_cross_u2_y = u1z[j_1]*u2x[j_2] - u1x[j_1]*u2z[j_2];
+                        Real u1_cross_u2_z = u1x[j_1]*u2y[j_2] - u1y[j_1]*u2x[j_2];
+
+                        Real v1_cross_v2_square = (u1_cross_u2_x*u1_cross_u2_x + u1_cross_u2_y*u1_cross_u2_y + u1_cross_u2_z*u1_cross_u2_z) / (gamma1*gamma1*gamma2*gamma2);
+
+                        Real radicand = v1_minus_v2_square - v1_cross_v2_square / c2;
+
+                        Real dL_dEcom = std::sqrt( radicand ) * w1[j_1] * w2[j_2] / dV / bin_size; // s^-1 m^-2 J^-1
```

Comment:
Given the expression of the luminosity:
$d\mathcal{L} = \int \sqrt{(\boldsymbol{\beta}_1 - \boldsymbol{\beta}_2)^2 - (\boldsymbol{\beta}_1 \times \boldsymbol{\beta}_2 )^2} f_1 f_2 d\boldsymbol{x} d\boldsymbol{v} dt$
I think that this should be multiplied by `dt`.

Accordingly, the results should be in `m^-2 . J^-1`, instead of `s^-1 m^-2 J^-1`.
It only becomes `s^-1 m^-2 J^-1` once we multiply by the repetition rate of the collider (i.e. number of collisions per second).